src/k2/module/k23si/client/k23si_client.h

```diff
@@ -285,7 +274,8 @@ class K2TxnHandle {
     }
 
     template <typename T1>
-    seastar::future<PartialUpdateResult> partialUpdate(T1& record, std::vector<k2::String> fieldsName) {
```

Comment:
I think it'd be better if we just gave overloads which take an extra dto::key. Then have the non-key API call into the key one.  same goes for the other changed APIs (make*, read, write)